### DISCUSSION TITLE
chalk does not produce colored logs if run with --no-worker-threads

### DISCUSSION THREAD

**Discussion by stevecaldwell77** [avajs/ava#3183] [Q&A] [answered]
With this setup:

- packages:
  - ava `5.2.0`
  - chalk `4.1.2`
  - ts-node `10.9.1`
  - typescript `4.9.5`
- Using `ts-node` for compilation
- TS configured to use module mode of `commonjs`

I'm trying to use chalk to color my log message, e.g.:

```ts
import test from "ava";
import chalk from "chalk";

test("colors are logged", (t) => {
  t.log(chalk.red("this message is in red"));
  t.log(chalk.green("this message is in green"));
  t.pass();
});
```

The messages show up correctly colored when I run ava normally:

```bash
npx ava
```

But if I turn off worker threads the messages end up gray:

```bash
npx ava --no-worker-threads
```

I have a sample repository here: https://github.com/stevecaldwell77/ava-chalk

And here is a StackBlitz link: https://stackblitz.com/github/stevecaldwell77/ava-chalk?file=README.md

**Comment by novemberborn** [comment, marked as answer]
Chalk detects color support based on its environment, my guess is that in a child process it doesn't inherit those details. You should be able to instantiate a Chalk instance that explicitly enables colors, or use environment variables or some of its other override options.

AVA does the same internally for its own Chalk instance.

**Reply by stevecaldwell77** [reply]
Yes, this works. Thanks!

```ts
import test from 'ava';
import { Instance as Chalk } from 'chalk';

const chalk = new Chalk({ level: 1 });

test('colors are logged', (t) => {
  t.log(chalk.red('this message is in red'));
  t.log(chalk.green('this message is in green'));
  t.pass();
});
```
